packages/trie/src/db/checkpoint.ts

```diff
@@ -17,6 +17,21 @@ export class CheckpointDB implements DB {
     this.checkpoints = []
   }
 
+  /**
+   * Flush the checkpoints and use the given checkpoints instead.
+   * @param {Checkpoint[]} checkpoints
+   */
+  setCheckpoints(checkpoints: Checkpoint[]) {
+    this.checkpoints = []
```

Comment:
I find this rather dangerous, should we not check if we currently are uncheckpointed (i.e. `this.checkpoints.length === 0`)?